app/src/main/java/org/mozilla/focus/widget/ProgressBar.kt

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.widget
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.arch.lifecycle.Observer
+import android.os.Handler
+import android.os.Looper
+import android.os.Message
+import android.widget.LinearLayout
+import android.widget.TextView
+import org.mozilla.focus.R
+import org.mozilla.focus.architecture.NonNullObserver
+import org.mozilla.focus.fragment.BrowserFragment
+import org.mozilla.focus.utils.GifImageView
+import java.lang.ref.WeakReference
+import java.util.concurrent.TimeUnit
+
+private const val HIDE_MESSAGE_ID = 0
+private const val HIDE_ANIMATION_DURATION_MILLIS = 250L
+private val HIDE_AFTER_MILLIS = TimeUnit.SECONDS.toMillis(3)
+
+class ProgressBar @JvmOverloads constructor(
```

Comment:
nit: There's a built-in Android class called ProgressBar - can we call this `FirefoxProgressBar`, or similar?